### ISSUE TITLE
Using formex_ecto with abstract tables

### ISSUE THREAD

**Issue by JohnKacz** [jakub-zawislak/formex_ecto#8]
Hi,

I'm trying to use `formex` and `formex_ecto` with a polymorphic association as discribed in the `ecto` docs [here](https://hexdocs.pm/ecto/2.2.6/Ecto.Schema.html#belongs_to/3-polymorphic-associations). 

When `update_form_data` is called after `create_changeset`, `@repo.insert` fails with
````
 Postgrex.Error at PUT /task/1
ERROR 42P01 (undefined_table) relation "abstract table: comments" does not exist

    query: INSERT INTO "abstract table: comments" (...)
````

Is there a way (either with `modify_changeset` or by explicitly setting `:struct_module`) to specify the correct table?

If not I'd be happy to work on PR to add the ability to check for abstract table which is stored in `__meta__` (i.e. `__meta__: #Ecto.Schema.Metadata<:built, "abstract table: comments">`) if you could give me some guidance on the best place to change it.

Finally I'm not actually using `has_many` but `has_one` but I don't think that matters as `formex_collection_child()` wouldn't change this issue.

**Comment by jakub-zawislak** [comment]
I just want to let you know that I already saw that post on elixir forum 😄 
If you want to try fix it, then look at `changeset.ex`, function `cast_embedded_forms`
I already found that in `builder.ex`, `preload_assocs` there is already working code that gets data for this polymorphic association. So only changeset doesn't work.

**Comment by JohnKacz** [comment]
Thanks Jakub. Yes that was me. So in my experiments I already discovered that if I create the polymorphic association manually then formex_ecto handles everything just fine. It's only when first creating a polymorphic association that doesn't work. Which is I think what you are saying about `preload_assocs`

> When using this technique, the only limitation is that you cannot build comments directly. For example, the command below
> 
> `Repo.insert!(%Comment{})`
> 
> will attempt to use the abstract table. Instead, one should use
> 
> `Repo.insert!(build_assoc(post, :comments))`

So `cast_embedded_forms` needs to determine if `build_assoc` should be used, but after that I'm not sure how to take the newly built associated Struct and add it to the changeset.

**Comment by JohnKacz** [comment]
Also is the idea of using `modify_changeset` (and maybe also `fields_casted_manually`) not possible?

**Comment by jakub-zawislak** [comment]
No, `modify_changeset` and so on is not the answer. 
If you create `%Comment{}` (`struct(Comment)`), then you see 
```elixir
%App.Comment{
  __meta__: #Ecto.Schema.Metadata<:built, "abstract table: comments">, # <- that weird name
```
But already existing comments from article look normal (the same is with the `build_assoc`):
```elixir
%App.Comment{
  __meta__: #Ecto.Schema.Metadata<:loaded, "article_comments">,
```
Internally I create a new struct by calling a `struct` function, in the main `formex` library. I should also made a struct by calling `build_assoc` in `formex_ecto` in case of polymorphic table. I will fix it tomorrow, now I'm going to sleep 😄

**Comment by jakub-zawislak** [comment]
Fixed, please update `formex_ecto` to `0.2.3`

**Comment by JohnKacz** [comment]
Brilliant! Thank you @jakub-zawislak
